client/transport/client.go

```diff
@@ -14,6 +21,7 @@ type Sender interface {
 
 // Client is an interface that abstracts all remote connections.
 type Client interface {
+	Dialer
```

Comment:
Actual dialers are configured by https://github.com/docker/go-connections/blob/master/sockets/sockets.go#L22.

There doesn't seem to be a good way to get to these without changing that project.

I think it is okay if this interface has a `Dialer() Dialer` method. Seems odd, but preserves the taxonomy here.